File: app/src/main/java/com/lemayn/review/MainActivity.java
package com.lemayn.review;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.lemayn.review.databinding.ActivityMainBinding;
import com.lemayn.review.jni.Hello;

import java.util.Arrays;

/**
 * author: ly
 * date  : 2019/3/18 22:43
 * desc  :
 */
public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private Hello hello;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        hello = new Hello(this);

        binding.text.setText(hello.getStringFromC());

        int[] array = {1, 2, 3, 4, 5};
        hello.encodeArray(array);
        binding.text.setText(Arrays.toString(array));
        for (int i : array) {
            System.out.println(i);
        }

        binding.text.postDelayed(new Runnable() {
            @Override
            public void run() {
                hello.cLog();
            }
        }, 3000);
    }

}

File: app/src/main/java/com/lemayn/review/MainActivity.kt
package com.lemayn.review

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem

class MainActivity : AppCompatActivity() {

    private lateinit var viewModel: ViewModel
    private lateinit var fm: FragmentManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
        fm = supportFragmentManager
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item?.itemId) {
            R.id.balance -> replaceFragment(item.itemId, BalanceFragment.getInstance())
        }
        return true
    }

    private fun replaceFragment(id: Int, fragment: Fragment) {
        fm.beginTransaction().replace(id, fragment).commit()
    }
}
